### ISSUE TITLE
Question on using reward from discriminator instead for policy update

### ISSUE THREAD

**Issue by guoyangqin** [ku2482/gail-airl-ppo.pytorch#10]
Nice, finally found this project that updates the policy using reward from discriminator and aligns itself with the algorithm in the GAIL paper. In many other libraries, they just use the reward from the environment. I was wondering why they do that and if optimizing policy with that reward detached from discriminator can really maximize the objective function.

**Comment by Charlesyyun** [comment]
Exactly, that's why we need GAIL! Because we don't know about the real reward function.
